### ISSUE TITLE
Cannot create user data directory

### ISSUE THREAD

**Issue by fcole90** [zloirock/core-js#915]
I was installing `expo-cli` as a global package, and core-js is a dependency. It failed installing correctly.
```
sudo npm install -g expo-cli --scripts-prepend-node-path=/snap/bin/node
npm WARN invalid config scripts-prepend-node-path="/snap/bin/node"
npm WARN deprecated @hapi/joi@17.1.1: Switch to 'npm install joi'
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated joi@11.4.0: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated topo@2.0.2: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated hoek@4.2.1: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated @hapi/formula@2.0.0: Moved to 'npm install @sideway/formula'
npm WARN deprecated @hapi/address@4.1.0: Moved to 'npm install @sideway/address'
npm WARN deprecated @hapi/pinpoint@2.0.0: Moved to 'npm install @sideway/pinpoint'
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
/usr/local/bin/expo -> /usr/local/lib/node_modules/expo-cli/bin/expo.js
/usr/local/bin/expo-cli -> /usr/local/lib/node_modules/expo-cli/bin/expo.js
npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/3790/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> core-js@2.6.12 postinstall /usr/local/lib/node_modules/expo-cli/node_modules/babel-runtime/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

2021/03/01 00:35:18.959395 cmd_run.go:994: WARNING: cannot create user data directory: cannot create "/nonexistent/snap/node/3790": mkdir /nonexistent: permission denied
cannot create user data directory: /nonexistent/snap/node/3790: Permission denied
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/expo-cli/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/expo-cli/node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/expo-cli/node_modules/webpack-dev-server/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN @pmmmwh/react-refresh-webpack-plugin@0.3.3 requires a peer of react-refresh@^0.8.2 but none is installed. You must install peer dependencies yourself.
npm WARN ajv-keywords@3.5.2 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! core-js@2.6.12 postinstall: `node -e "try{require('./postinstall')}catch(e){}"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the core-js@2.6.12 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-02-28T22_35_19_895Z-debug.log
```

My system is Ubuntu 20.04 and I'm using [node as a snap](https://github.com/nodejs/snap).
I installed the snap with the node and npm environment with `sudo snap install node --classic --channel=14` if it matters.

**Comment by fcole90** [comment]
The content of the mentioned debug file, `/root/.npm/_logs/2021-02-28T22_35_19_895Z-debug.log`, is as follows:

```
[truncated for brevity]
46947 silly postinstall uuid@3.3.2
46948 info lifecycle uuid@3.3.2~postinstall: uuid@3.3.2
46949 silly postinstall xmldom@0.1.27
46950 info lifecycle xmldom@0.1.27~postinstall: xmldom@0.1.27
46951 silly postinstall yallist@3.1.1
46952 info lifecycle yallist@3.1.1~postinstall: yallist@3.1.1
46953 silly postinstall @hapi/formula@2.0.0
46954 info lifecycle @hapi/formula@2.0.0~postinstall: @hapi/formula@2.0.0
46955 silly postinstall @hapi/hoek@9.1.1
46956 info lifecycle @hapi/hoek@9.1.1~postinstall: @hapi/hoek@9.1.1
46957 silly postinstall @hapi/address@4.1.0
46958 info lifecycle @hapi/address@4.1.0~postinstall: @hapi/address@4.1.0
46959 silly postinstall @hapi/pinpoint@2.0.0
46960 info lifecycle @hapi/pinpoint@2.0.0~postinstall: @hapi/pinpoint@2.0.0
46961 silly postinstall @hapi/topo@5.0.0
46962 info lifecycle @hapi/topo@5.0.0~postinstall: @hapi/topo@5.0.0
46963 silly postinstall @hapi/joi@17.1.1
46964 info lifecycle @hapi/joi@17.1.1~postinstall: @hapi/joi@17.1.1
46965 silly postinstall color-name@1.1.4
46966 info lifecycle color-name@1.1.4~postinstall: color-name@1.1.4
46967 silly postinstall color-convert@2.0.1
46968 info lifecycle color-convert@2.0.1~postinstall: color-convert@2.0.1
46969 silly postinstall ansi-styles@4.3.0
46970 info lifecycle ansi-styles@4.3.0~postinstall: ansi-styles@4.3.0
46971 silly postinstall has-flag@4.0.0
46972 info lifecycle has-flag@4.0.0~postinstall: has-flag@4.0.0
46973 silly postinstall supports-color@7.2.0
46974 info lifecycle supports-color@7.2.0~postinstall: supports-color@7.2.0
46975 silly postinstall chalk@3.0.0
46976 info lifecycle chalk@3.0.0~postinstall: chalk@3.0.0
46977 silly postinstall @nodelib/fs.stat@2.0.4
46978 info lifecycle @nodelib/fs.stat@2.0.4~postinstall: @nodelib/fs.stat@2.0.4
46979 silly postinstall @npmcli/ci-detect@1.3.0
46980 info lifecycle @npmcli/ci-detect@1.3.0~postinstall: @npmcli/ci-detect@1.3.0
46981 silly postinstall mkdirp@1.0.4
46982 info lifecycle mkdirp@1.0.4~postinstall: mkdirp@1.0.4
46983 silly postinstall mkdirp@1.0.4
46984 info lifecycle mkdirp@1.0.4~postinstall: mkdirp@1.0.4
46985 silly postinstall @npmcli/node-gyp@1.0.2
46986 info lifecycle @npmcli/node-gyp@1.0.2~postinstall: @npmcli/node-gyp@1.0.2
46987 silly postinstall ultron@1.0.2
46988 info lifecycle ultron@1.0.2~postinstall: ultron@1.0.2
46989 silly postinstall color-name@1.1.4
46990 info lifecycle color-name@1.1.4~postinstall: color-name@1.1.4
46991 silly postinstall color-convert@2.0.1
46992 info lifecycle color-convert@2.0.1~postinstall: color-convert@2.0.1
46993 silly postinstall ansi-styles@4.3.0
46994 info lifecycle ansi-styles@4.3.0~postinstall: ansi-styles@4.3.0
46995 silly postinstall has-flag@4.0.0
46996 info lifecycle has-flag@4.0.0~postinstall: has-flag@4.0.0
46997 silly postinstall supports-color@7.2.0
46998 info lifecycle supports-color@7.2.0~postinstall: supports-color@7.2.0
46999 silly postinstall chalk@3.0.0
47000 info lifecycle chalk@3.0.0~postinstall: chalk@3.0.0
47001 silly postinstall @sindresorhus/is@0.14.0
47002 info lifecycle @sindresorhus/is@0.14.0~postinstall: @sindresorhus/is@0.14.0
47003 silly postinstall @tootallnate/once@1.1.2
47004 info lifecycle @tootallnate/once@1.1.2~postinstall: @tootallnate/once@1.1.2
47005 silly postinstall @types/anymatch@1.3.1
47006 info lifecycle @types/anymatch@1.3.1~postinstall: @types/anymatch@1.3.1
47007 silly postinstall @types/html-minifier-terser@5.1.1
47008 info lifecycle @types/html-minifier-terser@5.1.1~postinstall: @types/html-minifier-terser@5.1.1
47009 silly postinstall @types/http-cache-semantics@4.0.0
47010 info lifecycle @types/http-cache-semantics@4.0.0~postinstall: @types/http-cache-semantics@4.0.0
47011 silly postinstall @types/istanbul-lib-coverage@2.0.3
47012 info lifecycle @types/istanbul-lib-coverage@2.0.3~postinstall: @types/istanbul-lib-coverage@2.0.3
47013 silly postinstall @types/istanbul-lib-report@3.0.0
47014 info lifecycle @types/istanbul-lib-report@3.0.0~postinstall: @types/istanbul-lib-report@3.0.0
47015 silly postinstall @types/istanbul-reports@1.1.2
47016 info lifecycle @types/istanbul-reports@1.1.2~postinstall: @types/istanbul-reports@1.1.2
47017 silly postinstall @types/json-schema@7.0.7
47018 info lifecycle @types/json-schema@7.0.7~postinstall: @types/json-schema@7.0.7
47019 silly postinstall @types/minimatch@3.0.3
47020 info lifecycle @types/minimatch@3.0.3~postinstall: @types/minimatch@3.0.3
47021 silly postinstall @types/node@9.6.61
47022 info lifecycle @types/node@9.6.61~postinstall: @types/node@9.6.61
47023 silly postinstall @types/glob@7.1.3
47024 info lifecycle @types/glob@7.1.3~postinstall: @types/glob@7.1.3
47025 silly postinstall @types/keyv@3.1.1
47026 info lifecycle @types/keyv@3.1.1~postinstall: @types/keyv@3.1.1
47027 silly postinstall @types/q@1.5.4
47028 info lifecycle @types/q@1.5.4~postinstall: @types/q@1.5.4
47029 silly postinstall @types/responselike@1.0.0
47030 info lifecycle @types/responselike@1.0.0~postinstall: @types/responselike@1.0.0
47031 silly postinstall @types/cacheable-request@6.0.1
47032 info lifecycle @types/cacheable-request@6.0.1~postinstall: @types/cacheable-request@6.0.1
47033 silly postinstall @types/retry@0.12.0
47034 info lifecycle @types/retry@0.12.0~postinstall: @types/retry@0.12.0
47035 silly postinstall @types/source-list-map@0.1.2
47036 info lifecycle @types/source-list-map@0.1.2~postinstall: @types/source-list-map@0.1.2
47037 silly postinstall @types/tapable@1.0.6
47038 info lifecycle @types/tapable@1.0.6~postinstall: @types/tapable@1.0.6
47039 silly postinstall @types/text-table@0.2.1
47040 info lifecycle @types/text-table@0.2.1~postinstall: @types/text-table@0.2.1
47041 silly postinstall source-map@0.6.1
47042 info lifecycle source-map@0.6.1~postinstall: source-map@0.6.1
47043 silly postinstall @types/uglify-js@3.12.0
47044 info lifecycle @types/uglify-js@3.12.0~postinstall: @types/uglify-js@3.12.0
47045 silly postinstall source-map@0.7.3
47046 info lifecycle source-map@0.7.3~postinstall: source-map@0.7.3
47047 silly postinstall @types/webpack-sources@2.1.0
47048 info lifecycle @types/webpack-sources@2.1.0~postinstall: @types/webpack-sources@2.1.0
47049 silly postinstall source-map@0.6.1
47050 info lifecycle source-map@0.6.1~postinstall: source-map@0.6.1
47051 silly postinstall @types/webpack@4.41.26
47052 info lifecycle @types/webpack@4.41.26~postinstall: @types/webpack@4.41.26
47053 silly postinstall @types/yargs-parser@20.2.0
47054 info lifecycle @types/yargs-parser@20.2.0~postinstall: @types/yargs-parser@20.2.0
47055 silly postinstall @types/yargs@15.0.13
47056 info lifecycle @types/yargs@15.0.13~postinstall: @types/yargs@15.0.13
47057 silly postinstall @jest/types@25.5.0
47058 info lifecycle @jest/types@25.5.0~postinstall: @jest/types@25.5.0
47059 silly postinstall @webassemblyjs/floating-point-hex-parser@1.9.0
47060 info lifecycle @webassemblyjs/floating-point-hex-parser@1.9.0~postinstall: @webassemblyjs/floating-point-hex-parser@1.9.0
47061 silly postinstall @webassemblyjs/helper-api-error@1.9.0
47062 info lifecycle @webassemblyjs/helper-api-error@1.9.0~postinstall: @webassemblyjs/helper-api-error@1.9.0
47063 silly postinstall @webassemblyjs/helper-buffer@1.9.0
47064 info lifecycle @webassemblyjs/helper-buffer@1.9.0~postinstall: @webassemblyjs/helper-buffer@1.9.0
47065 silly postinstall @webassemblyjs/helper-fsm@1.9.0
47066 info lifecycle @webassemblyjs/helper-fsm@1.9.0~postinstall: @webassemblyjs/helper-fsm@1.9.0
47067 silly postinstall @webassemblyjs/helper-wasm-bytecode@1.9.0
47068 info lifecycle @webassemblyjs/helper-wasm-bytecode@1.9.0~postinstall: @webassemblyjs/helper-wasm-bytecode@1.9.0
47069 silly postinstall @webassemblyjs/utf8@1.9.0
47070 info lifecycle @webassemblyjs/utf8@1.9.0~postinstall: @webassemblyjs/utf8@1.9.0
47071 silly postinstall @xtuc/ieee754@1.2.0
47072 info lifecycle @xtuc/ieee754@1.2.0~postinstall: @xtuc/ieee754@1.2.0
47073 silly postinstall @webassemblyjs/ieee754@1.9.0
47074 info lifecycle @webassemblyjs/ieee754@1.9.0~postinstall: @webassemblyjs/ieee754@1.9.0
47075 silly postinstall @xtuc/long@4.2.2
47076 info lifecycle @xtuc/long@4.2.2~postinstall: @xtuc/long@4.2.2
47077 silly postinstall @webassemblyjs/leb128@1.9.0
47078 info lifecycle @webassemblyjs/leb128@1.9.0~postinstall: @webassemblyjs/leb128@1.9.0
47079 silly postinstall @webassemblyjs/wast-printer@1.9.0
47080 info lifecycle @webassemblyjs/wast-printer@1.9.0~postinstall: @webassemblyjs/wast-printer@1.9.0
47081 silly postinstall @webassemblyjs/helper-code-frame@1.9.0
47082 info lifecycle @webassemblyjs/helper-code-frame@1.9.0~postinstall: @webassemblyjs/helper-code-frame@1.9.0
47083 silly postinstall @webassemblyjs/wast-parser@1.9.0
47084 info lifecycle @webassemblyjs/wast-parser@1.9.0~postinstall: @webassemblyjs/wast-parser@1.9.0
47085 silly postinstall @webassemblyjs/ast@1.9.0
47086 info lifecycle @webassemblyjs/ast@1.9.0~postinstall: @webassemblyjs/ast@1.9.0
47087 silly postinstall @webassemblyjs/helper-module-context@1.9.0
47088 info lifecycle @webassemblyjs/helper-module-context@1.9.0~postinstall: @webassemblyjs/helper-module-context@1.9.0
47089 silly postinstall @webassemblyjs/wasm-gen@1.9.0
47090 info lifecycle @webassemblyjs/wasm-gen@1.9.0~postinstall: @webassemblyjs/wasm-gen@1.9.0
47091 silly postinstall @webassemblyjs/helper-wasm-section@1.9.0
47092 info lifecycle @webassemblyjs/helper-wasm-section@1.9.0~postinstall: @webassemblyjs/helper-wasm-section@1.9.0
47093 silly postinstall @webassemblyjs/wasm-parser@1.9.0
47094 info lifecycle @webassemblyjs/wasm-parser@1.9.0~postinstall: @webassemblyjs/wasm-parser@1.9.0
47095 silly postinstall @webassemblyjs/wasm-opt@1.9.0
47096 info lifecycle @webassemblyjs/wasm-opt@1.9.0~postinstall: @webassemblyjs/wasm-opt@1.9.0
47097 silly postinstall @webassemblyjs/wasm-edit@1.9.0
47098 info lifecycle @webassemblyjs/wasm-edit@1.9.0~postinstall: @webassemblyjs/wasm-edit@1.9.0
47099 silly postinstall abbrev@1.1.1
47100 info lifecycle abbrev@1.1.1~postinstall: abbrev@1.1.1
47101 silly postinstall acorn@6.4.2
47102 info lifecycle acorn@6.4.2~postinstall: acorn@6.4.2
47103 silly postinstall address@1.1.2
47104 info lifecycle address@1.1.2~postinstall: address@1.1.2
47105 silly postinstall ajv-errors@1.0.1
47106 info lifecycle ajv-errors@1.0.1~postinstall: ajv-errors@1.0.1
47107 silly postinstall ajv-keywords@3.5.2
47108 info lifecycle ajv-keywords@3.5.2~postinstall: ajv-keywords@3.5.2
47109 silly postinstall alphanum-sort@1.0.2
47110 info lifecycle alphanum-sort@1.0.2~postinstall: alphanum-sort@1.0.2
47111 silly postinstall uuid@3.4.0
47112 info lifecycle uuid@3.4.0~postinstall: uuid@3.4.0
47113 silly postinstall ansi-regex@4.1.0
47114 info lifecycle ansi-regex@4.1.0~postinstall: ansi-regex@4.1.0
47115 silly postinstall emoji-regex@7.0.3
47116 info lifecycle emoji-regex@7.0.3~postinstall: emoji-regex@7.0.3
47117 silly postinstall is-fullwidth-code-point@2.0.0
47118 info lifecycle is-fullwidth-code-point@2.0.0~postinstall: is-fullwidth-code-point@2.0.0
47119 silly postinstall strip-ansi@5.2.0
47120 info lifecycle strip-ansi@5.2.0~postinstall: strip-ansi@5.2.0
47121 silly postinstall string-width@3.1.0
47122 info lifecycle string-width@3.1.0~postinstall: string-width@3.1.0
47123 silly postinstall ansi-align@3.0.0
47124 info lifecycle ansi-align@3.0.0~postinstall: ansi-align@3.0.0
47125 silly postinstall ansi-colors@3.2.4
47126 info lifecycle ansi-colors@3.2.4~postinstall: ansi-colors@3.2.4
47127 silly postinstall ansi-escapes@3.2.0
47128 info lifecycle ansi-escapes@3.2.0~postinstall: ansi-escapes@3.2.0
47129 silly postinstall ansi-html@0.0.7
47130 info lifecycle ansi-html@0.0.7~postinstall: ansi-html@0.0.7
47131 silly postinstall ansi-regex@5.0.0
47132 info lifecycle ansi-regex@5.0.0~postinstall: ansi-regex@5.0.0
47133 silly postinstall any-base@1.1.0
47134 info lifecycle any-base@1.1.0~postinstall: any-base@1.1.0
47135 silly postinstall any-promise@1.3.0
47136 info lifecycle any-promise@1.3.0~postinstall: any-promise@1.3.0
47137 silly postinstall application-config-path@0.1.0
47138 info lifecycle application-config-path@0.1.0~postinstall: application-config-path@0.1.0
47139 silly postinstall aproba@1.2.0
47140 info lifecycle aproba@1.2.0~postinstall: aproba@1.2.0
47141 silly postinstall arr-diff@4.0.0
47142 info lifecycle arr-diff@4.0.0~postinstall: arr-diff@4.0.0
47143 silly postinstall arr-flatten@1.1.0
47144 info lifecycle arr-flatten@1.1.0~postinstall: arr-flatten@1.1.0
47145 silly postinstall arr-union@3.1.0
47146 info lifecycle arr-union@3.1.0~postinstall: arr-union@3.1.0
47147 silly postinstall array-filter@1.0.0
47148 info lifecycle array-filter@1.0.0~postinstall: array-filter@1.0.0
47149 silly postinstall array-flatten@1.1.1
47150 info lifecycle array-flatten@1.1.1~postinstall: array-flatten@1.1.1
47151 silly postinstall array-map@0.0.0
47152 info lifecycle array-map@0.0.0~postinstall: array-map@0.0.0
47153 silly postinstall array-reduce@0.0.0
47154 info lifecycle array-reduce@0.0.0~postinstall: array-reduce@0.0.0
47155 silly postinstall array-uniq@1.0.3
47156 info lifecycle array-uniq@1.0.3~postinstall: array-uniq@1.0.3
47157 silly postinstall array-union@1.0.2
47158 info lifecycle array-union@1.0.2~postinstall: array-union@1.0.2
47159 silly postinstall array-unique@0.3.2
47160 info lifecycle array-unique@0.3.2~postinstall: array-unique@0.3.2
47161 silly postinstall arrify@2.0.1
47162 info lifecycle arrify@2.0.1~postinstall: arrify@2.0.1
47163 silly postinstall bn.js@4.12.0
47164 info lifecycle bn.js@4.12.0~postinstall: bn.js@4.12.0
47165 silly postinstall assert-plus@1.0.0
47166 info lifecycle assert-plus@1.0.0~postinstall: assert-plus@1.0.0
47167 silly postinstall inherits@2.0.1
47168 info lifecycle inherits@2.0.1~postinstall: inherits@2.0.1
47169 silly postinstall util@0.10.3
47170 info lifecycle util@0.10.3~postinstall: util@0.10.3
47171 silly postinstall assign-symbols@1.0.0
47172 info lifecycle assign-symbols@1.0.0~postinstall: assign-symbols@1.0.0
47173 silly postinstall async@1.5.2
47174 info lifecycle async@1.5.2~postinstall: async@1.5.2
47175 silly postinstall async-each@1.0.3
47176 info lifecycle async-each@1.0.3~postinstall: async-each@1.0.3
47177 silly postinstall async-limiter@1.0.1
47178 info lifecycle async-limiter@1.0.1~postinstall: async-limiter@1.0.1
47179 silly postinstall asynckit@0.4.0
47180 info lifecycle asynckit@0.4.0~postinstall: asynckit@0.4.0
47181 silly postinstall at-least-node@1.0.0
47182 info lifecycle at-least-node@1.0.0~postinstall: at-least-node@1.0.0
47183 silly postinstall atob@2.1.2
47184 info lifecycle atob@2.1.2~postinstall: atob@2.1.2
47185 silly postinstall available-typed-arrays@1.0.2
47186 info lifecycle available-typed-arrays@1.0.2~postinstall: available-typed-arrays@1.0.2
47187 silly postinstall aws-sign2@0.7.0
47188 info lifecycle aws-sign2@0.7.0~postinstall: aws-sign2@0.7.0
47189 silly postinstall aws4@1.11.0
47190 info lifecycle aws4@1.11.0~postinstall: aws4@1.11.0
47191 silly postinstall semver@6.3.0
47192 info lifecycle semver@6.3.0~postinstall: semver@6.3.0
47193 silly postinstall babel-plugin-syntax-object-rest-spread@6.13.0
47194 info lifecycle babel-plugin-syntax-object-rest-spread@6.13.0~postinstall: babel-plugin-syntax-object-rest-spread@6.13.0
47195 silly postinstall babel-plugin-syntax-trailing-function-commas@7.0.0-beta.0
47196 info lifecycle babel-plugin-syntax-trailing-function-commas@7.0.0-beta.0~postinstall: babel-plugin-syntax-trailing-function-commas@7.0.0-beta.0
47197 silly postinstall core-js@2.6.12
47198 info lifecycle core-js@2.6.12~postinstall: core-js@2.6.12
47199 warn lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/3790/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
47200 verbose lifecycle core-js@2.6.12~postinstall: unsafe-perm in lifecycle false
47201 verbose lifecycle core-js@2.6.12~postinstall: PATH: /snap/node/3790/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/usr/local/lib/node_modules/expo-cli/node_modules/babel-runtime/node_modules/core-js/node_modules/.bin:/usr/local/lib/node_modules/expo-cli/node_modules/babel-runtime/node_modules/.bin:/usr/local/lib/node_modules/expo-cli/node_modules/.bin:/usr/local/lib/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
47202 verbose lifecycle core-js@2.6.12~postinstall: CWD: /usr/local/lib/node_modules/expo-cli/node_modules/babel-runtime/node_modules/core-js
47203 silly lifecycle core-js@2.6.12~postinstall: Args: [ '-c', `node -e "try{require('./postinstall')}catch(e){}"` ]
47204 silly lifecycle core-js@2.6.12~postinstall: Returned: code: 1  signal: null
47205 info lifecycle core-js@2.6.12~postinstall: Failed to exec postinstall script
47206 timing action:postinstall Completed in 123ms
47207 verbose unlock done using /root/.npm/_locks/staging-3a08f0df5026584d.lock for /usr/local/lib/node_modules/.staging
47208 timing stage:rollbackFailedOptional Completed in 763ms
47209 timing stage:runTopLevelLifecycles Completed in 36484ms
47210 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/expo-cli/node_modules/chokidar/node_modules/fsevents):
47211 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
47212 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
47212 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
47212 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
47212 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
47213 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/expo-cli/node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
47214 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
47215 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
47215 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
47215 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
47215 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
47216 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/expo-cli/node_modules/webpack-dev-server/node_modules/chokidar/node_modules/fsevents):
47217 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
47218 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
47218 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
47218 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
47218 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
47219 warn @pmmmwh/react-refresh-webpack-plugin@0.3.3 requires a peer of react-refresh@^0.8.2 but none is installed. You must install peer dependencies yourself.
47220 warn ajv-keywords@3.5.2 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.
47221 verbose stack Error: core-js@2.6.12 postinstall: `node -e "try{require('./postinstall')}catch(e){}"`
47221 verbose stack Exit status 1
47221 verbose stack     at EventEmitter.<anonymous> (/snap/node/3790/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
47221 verbose stack     at EventEmitter.emit (events.js:315:20)
47221 verbose stack     at ChildProcess.<anonymous> (/snap/node/3790/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
47221 verbose stack     at ChildProcess.emit (events.js:315:20)
47221 verbose stack     at maybeClose (internal/child_process.js:1048:16)
47221 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
47222 verbose pkgid core-js@2.6.12
47223 verbose cwd /home/fabio/Projects/aalto/fullstack_open_2020_part10
47224 verbose Linux 5.4.0-66-generic
47225 verbose argv "/snap/node/3790/bin/node" "/snap/node/3790/bin/npm" "install" "-g" "expo-cli" "--scripts-prepend-node-path=/snap/bin/node"
47226 verbose node v14.16.0
47227 verbose npm  v6.14.11
47228 error code ELIFECYCLE
47229 error errno 1
47230 error core-js@2.6.12 postinstall: `node -e "try{require('./postinstall')}catch(e){}"`
47230 error Exit status 1
47231 error Failed at the core-js@2.6.12 postinstall script.
47231 error This is probably not a problem with npm. There is likely additional logging output above.
47232 verbose exit [ 1, true ]
```

**Comment by zloirock** [comment]
I'm not familiar with `snap`, but fast googling shows that it's a `snap` permissions issue.

**Comment by fcole90** [comment]
Can you link to what you found? 

I'm familiar with the snap format and I'm willing to help to debug this issue. Snap is like a frozen container (like an immutable Docker), but node modules are installed in a writable portion of the system, like `/usr/lib/node_modules`, so I'm not sure how this relates to snap.

What was the postinstall doing when it crashed? This can be very useful to debug the snap packaging system or the node snap :blush:

**Comment by zloirock** [comment]
@fcole90
> Can you link to what you found?

the first link from search https://github.com/admc/wd/issues/586

> What was the postinstall doing when it crashed?

it does not create any folder, maximally it could create a temporal file in the default directory for temporal files.

**Comment by fcole90** [comment]
Thanks for sharing, I added some comments there too. 

What is the default directory for temporal files? Maybe this is an issue on how node is configured in the snap :thinking:

**Comment by zloirock** [comment]
```js
require('os').tmpdir()
```

**Comment by fcole90** [comment]
Thanks, I'll make some experiment and update you :blush:

**Comment by fcole90** [comment]
I can't understand the issue, as `tmpdir()` returns `/tmp`, which is valid. I also tried running the [postinstall script](https://github.com/zloirock/core-js/blob/master/packages/core-js/postinstall.js) (is it the correct one?) and I didn't obtain any error. I also tried to reinstall `expo-cli`, but I have now configured a [local npm directory](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally), and the installation worked correctly.

I think the bug is not in your code, it seems something related to how npm handles installations in snaps. I'll try to ask more ideas on the  snap forum :blush: Thanks for your time and your library :grinning:
